src/native/fs/fs_napi.cpp

```diff
@@ -2115,6 +2127,32 @@ set_debug_level(const Napi::CallbackInfo& info)
     return info.Env().Undefined();
 }
 
+/**
+ * register noobaa args to GPFS
+ */
+static Napi::Value
+register_gpfs_noobaa(const Napi::CallbackInfo& info)
+{
+    Napi::Object params = info[0].As<Napi::Object>();
+    struct gpfs_ganesha_noobaa_arg args = {
+        params.Get("version").ToNumber(),
+        params.Get("delay").ToNumber(),
+        params.Get("flags").ToNumber(),
```

Comment:
can we use explicit named fields in this struct initializer - this is better because it avoids relying on the struct field order
```suggestion
        .noobaa_version = params.Get("version").ToNumber(),
        .noobaa_delay  = params.Get("delay").ToNumber(),
        .noobaa_flags = params.Get("flags").ToNumber(),
```

Reply:
@guymguym I tried it before but removed it because it's not compiling on our CI - 
```
../src/native/fs/fs_napi.cpp: In function ‘Napi::Value noobaa::register_gpfs_noobaa(const Napi::CallbackInfo&)’:
../src/native/fs/fs_napi.cpp:2138:9: error: C++ designated initializers only available with ‘-std=c++20’ or ‘-std=gnu++20’ [-Werror=pedantic]
 2138 |         .noobaa_version = params.Get("version").ToNumber(),
      |         ^
../src/native/fs/fs_napi.cpp:2139:9: error: C++ designated initializers only available with ‘-std=c++20’ or ‘-std=gnu++20’ [-Werror=pedantic]
 2139 |         .noobaa_delay = params.Get("delay").ToNumber(),
      |         ^
../src/native/fs/fs_napi.cpp:2140:9: error: C++ designated initializers only available with ‘-std=c++20’ or ‘-std=gnu++20’ [-Werror=pedantic]
 2140 |         .noobaa_flags = params.Get("flags").ToNumber(),
      |         ^
cc1plus: all warnings being treated as errors
make: *** [src/native/nb_native.target.mk:157: Release/obj.target/nb_native/src/native/fs/fs_napi.o] Error 1
```

Reply:
ah I see. perhaps it's a good time to bump up from c++17 to c++20 in common.gypi (notice it appears twice for both for linux and mac).